nixos/tests/teleport.nix

```diff
@@ -0,0 +1,86 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "teleport";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ ymatsiuk ];
+  };
+
+  nodes = {
+    client = { config, pkgs, ... }: {
+      services.teleport = {
+        enable = true;
+        settings = {
+          teleport = {
+            nodename = "client";
+            advertise_ip = "192.168.1.20";
+            auth_token = "8d1957b2-2ded-40e6-8297-d48156a898a9";
+            auth_servers = [ "192.168.1.10:3025" ];
+          };
+          ssh_service.enabled = true;
+          proxy_service.enabled = false;
+          auth_service.enabled = false;
+        };
+        debug.enable = true;
+      };
+      networking.interfaces.eth1.ipv4.addresses = [{
+        address = "192.168.1.20";
+        prefixLength = 24;
+      }];
+    };
+    server = { config, pkgs, ... }: {
+      services.teleport = {
+        enable = true;
+        settings = {
+          teleport = {
+            nodename = "server";
+            advertise_ip = "192.168.1.10";
+          };
+          ssh_service.enabled = true;
+          proxy_service.enabled = true;
+          auth_service = {
+            enabled = true;
+            tokens = [ "node:8d1957b2-2ded-40e6-8297-d48156a898a9" ];
+          };
+        };
+        diag.enable = true;
+        extraOpts = [
+          "--labels=RandomLabelKey=RandomLabelValue"
+        ];
+      };
+      environment.systemPackages = [ pkgs.jq ];
+      networking.firewall.allowedTCPPorts = [ 3025 ];
+      networking.interfaces.eth1.ipv4.addresses = [{
+        address = "192.168.1.10";
+        prefixLength = 24;
+      }];
+    };
+    standalone = { config, pkgs, ... }: {
+      services.teleport.enable = true;
+    };
+  };
+
+  testScript = ''
+    with subtest("start client after server is inititalized and accepting connections"):
+        server.wait_for_open_port("3025")
+        client.wait_for_open_port("3022")
```

Comment:
Seems like you could boot these simultaneously, with `client.start()`

(that way you have both booting at the same time, which is a bit faster)

Reply:
```suggestion
    with subtest("simultaneously boot client and server"):
        client.start()
        server.wait_for_open_port("3025")
        client.wait_for_open_port("3022")
```

Reply:
> Seems like you could boot these simultaneously, with `client.start()`
> 
> (that way you have both booting at the same time, which is a bit faster)

good one! I just realized that I can do `start_all()` to boot three VMs in parallel

Reply:
The problem with start_all() here is that you're using memory for the third vm while you aren't gonna use it yet. Not a huge deal in this case, but could be costly in other cases.

(that's also why I suggested splitting it out)

Reply:
I got your point now! 👍🏻